node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java

```diff
@@ -385,6 +385,18 @@ private NodeAcl getNodeAcl(Node node, NodeList candidates) {
         return List.of(getNodeAcl(node, candidates));
     }
 
+    /**
+     * Returns whether the zone managed by this node repository seems to be working.
+     * If too many nodes are not responding, there is probably some zone-wide issue
+     * and we should probably refrain from making changes to it.
+     */
+    public boolean isWorking() {
+        NodeList activeNodes = list(State.active);
+        if (activeNodes.size() <= 5) return true; // Not enough data to decide
+        NodeList downNodes = activeNodes.down();
+        return ! ( (double)downNodes.size() / (double)activeNodes.size() > 0.2 );
```

Comment:
Consider simplifying to `(double)downNodes.size() / (double)activeNodes.size() <= 0.2`.

Reply:
I find "working if not more than 20% fails" is easier to get than the simpler one but that might just be me ...